Sources/Tracing/Tracer.swift

```diff
@@ -350,6 +350,53 @@ public func withSpan<T, Instant: TracerInstant>(
     }
 }
 
+#if swift(>=5.7.0)
+/// Start a new ``Span`` and automatically end when the `operation` completes,
+/// including recording the `error` in case the operation throws.
+///
+/// The current task-local `ServiceContext` is picked up and provided to the underlying tracer.
+/// It is also possible to pass a specific `context` explicitly, in which case attempting
+/// to pick up the task-local context is prevented. This can be useful when we know that
+/// we're about to start a top-level span, or if a span should be started from a different,
+/// stored away previously,
+///
+/// - Warning: You MUST NOT ``Span/end()`` the span explicitly, because at the end of the `withSpan`
+///   operation closure returning the span will be closed automatically.
+///
+/// - Parameters:
+///   - operationName: The name of the operation being traced. This may be a handler function, database call, ...
+///   - context: The `ServiceContext` providing information on where to start the new ``Span``.
+///   - kind: The ``SpanKind`` of the new ``Span``.
+///   - function: The function name in which the span was started
+///   - fileID: The `fileID` where the span was started.
+///   - line: The file line where the span was started.
+///   - operation: The operation that this span should be measuring
+/// - Returns: the value returned by `operation`
+/// - Throws: the error the `operation` has thrown (if any)
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *) // for TaskLocal ServiceContext
+@preconcurrency
+public func withSpan<T>(
+    _ operationName: String,
+    context: @Sendable @autoclosure () -> ServiceContext = .current ?? .topLevel,
+    ofKind kind: SpanKind = .internal,
+    function: String = #function,
+    file fileID: String = #fileID,
+    line: UInt = #line,
+    _ operation: @Sendable (any Span) async throws -> T
```

Comment:
This must not be @Sendable -- see comment in test what this breaks then (e.g. inside an actor)

Reply:
Reverted.